grails-app/views/dashboard/megamenu.gsp

```diff
@@ -898,5 +596,48 @@
             </g:isUserInRole>
         </g:if>
      </g:hideIfIsNonInventoryManagedAndCanSubmitRequest>
+<g:hiddenField id="megamenu" name="megamenu" value="${megamenuConfig}"/>
+<g:hiddenField id="dashboardMenuItems" name="dashboardMenuItems" value="${megamenuConfig.dashboard.href}"/>
+<g:hiddenField id="inventoryMenuItems" name="inventoryMenuItems" value="${megamenuConfig.inventory.subsections}"/>
+<g:hiddenField id="purchasingMenuItems" name="purchasingMenuItems" value="${megamenuConfig.purchasing.subsections}"/>
+<g:hiddenField id="invoicingMenuItems" name="invoicingMenuItems" value="${megamenuConfig.invoicing.subsections}"/>
+<g:hiddenField id="inboundMenuItems" name="inboundMenuItems" value="${megamenuConfig.inbound.subsections}"/>
+<g:hiddenField id="outboundMenuItems" name="outboundMenuItems" value="${megamenuConfig.outbound.subsections}"/>
+<g:hiddenField id="reportingMenuItems" name="reportingMenuItems" value="${megamenuConfig.reporting.subsections}"/>
+<g:hiddenField id="productsMenuItems" name="productsMenuItems" value="${megamenuConfig.products.subsections}"/>
+<g:hiddenField id="stocklistMenuItems" name="stocklistMenuItems" value="${megamenuConfig.requisitionTemplate.menuItems}"/>
+<g:hiddenField id="currentLocation" name="currentLocation" value="${session?.warehouse?.id}"/>
 </ul>
 <!--MegaMenu Ends-->
+<script type="text/javascript">
+  $(document).ready(function() {
+    const currentLocationId = $("#currentLocation").val();
```

Comment:
Add a comment here that the code below is for checking the `active` state of menu items. Or move the code into a function named properly and call it here.